parallel_pick_behaviors/include/parallel_pick_behaviors/push_to_solution_queue.hpp

```diff
@@ -0,0 +1,51 @@
+// Copyright 2022 PickNik Inc.
+// All rights reserved.
+//
+// Unauthorized copying of this code base via any medium is strictly prohibited.
+// Proprietary and confidential.
+
+#pragma once
+
+#include <moveit_studio_behavior_interface/async_behavior_base.hpp>
+
+namespace moveit_studio::behaviors
+{
+/**
+ * @brief Push a new solution to the solution queue
+ *
+ * @details
+ * | Data Port Name   | Port Type        | Object Type                                             |
+ * |------------------|------------------|---------------------------------------------------------|
+ * | solution         | Input            | moveit_task_constructor_msgs::msg::Solution             |
+ * | solution queue   | Input/Output     | std::queue<moveit_task_constructor_msgs::msg::Solution> |
+ */
+class PushToSolutionQueue final : public AsyncBehaviorBase
```

Comment:
Could this be a regular `BT::SyncActionNode`? As currently written, the stuff in `doWork` don't look like they'll block, so the async functionality is probably overkill.

Reply:
At first I thought so too but [SyncActionNode](http://docs.ros.org/en/kinetic/api/behaviortree_cpp_v3/html/classBT_1_1SyncActionNode.html) cannot return RUNNING, that's why I decided to use that one. Alternatively, I could use an ActionBaseNode but then I'd need to implement every function

Reply:
We talked about this in-person: the push to the queue can by a SyncActionNode because it doesn't need to block. Popping from the queue needs to be something different (and not an AsyncBehaviorBase) because it needs to be able to return RUNNING while waiting for something to go into the queue.